cmd/clusterctl/client/cluster/proxy.go

```diff
@@ -70,9 +70,18 @@ func (k *proxy) NewClient() (client.Client, error) {
 		return nil, err
 	}
 
-	c, err := client.New(config, client.Options{Scheme: Scheme})
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to connect to the management cluster")
+	var c client.Client
+	// Nb. The operation is wrapped in a retry loop to make newClientSet more resilient to temporary connection problems.
+	connectBackoff := newConnectBackoff()
+	if err := retryWithExponentialBackoff(connectBackoff, func() error {
+		var err2 error
```

Comment:
```suggestion
		var err error
```

no need for 2, it'll shadow the outer one

Reply:
done